### PR TITLE
Add Login / SignUp Screens

### PR DESCRIPTION
**Pull request by omar-sarfraz**
This PR contains the following additions,
- NavBar
- Login Page
- Sign Up Page

### CODE REVIEW

File: React Project/src/pages/Login/Login.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                <div className="flex flex-col items-center">
+                    <h2 className="text-3xl font-medium mb-2">Login</h2>
+                    <p>Login to access thousands of projects</p>
+                </div>
+                <div className="my-10">
+                    <div className="flex flex-col gap-1">
+                        <label htmlFor="email" className="text-xl">
+                            Email
+                        </label>
+                        <input
```
[the remark this answers is not in the document]
Comment (omar-sarfraz) [reply React Project/src/pages/Login/Login.tsx]: We can also make component of the button

File: React Project/src/pages/Login/Login.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            return;
+        }
+
+        if (!password) {
+            setError({ type: "PASSWORD", message: "Password is required!" });
+            return;
+        }
+
+        const key = parseInt(import.meta.env.VITE_CIPHER_KEY);
+        const encryptedEmail = encrypt(key, email);
```
Comment (larebsyed) [inline React Project/src/pages/Login/Login.tsx:33]: why aren't you using the cezar-cypher package you created here?

File: React Project/src/pages/SignUp/SignUp.tsx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+            setError({ type: "CONFIRM_PASSWORD", message: "Please confirm your password!" });
+            return;
+        }
+
+        if (password !== confirmPassword) {
+            setError({ type: "CONFIRM_PASSWORD", message: "Passwords do not match!" });
+            return;
+        }
+
+        const key = parseInt(import.meta.env.VITE_CIPHER_KEY);
```
Comment (larebsyed) [inline React Project/src/pages/SignUp/SignUp.tsx:61]: what happens if the cypher key isn't present.

File: React Project/src/components/NotFound.tsx [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export default function NotFound() {
```
Comment (larebsyed) [inline React Project/src/components/NotFound.tsx:1]: This should be in pages

File: React Project/src/pages/SignUp/SignUp.tsx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+            lastName: encryptedLastName,
+            email: encryptedEmail,
+            password: encryptedPassword,
+        };
+        let existingUser = localStorage.getItem(encryptedEmail);
+
+        if (existingUser) alert("User with this email already exists");
+        else {
+            localStorage.setItem(encryptedEmail, JSON.stringify(userData));
+            alert("Account registered successfully");
```
Comment (larebsyed) [inline React Project/src/pages/SignUp/SignUp.tsx:81]: Please create a custom Alert component that is async, the browser alert are intrusive and disliked by users.

File: React Project/src/pages/Login/Login.tsx (hunk 2 above)
Comment (omar-sarfraz) [reply React Project/src/pages/Login/Login.tsx:33]: I've used the caesar-cipher package to encrypt name and email. For passwords, i am creating an hash as it is more secure. The encrypt function is from the caesar-cipher package.

File: React Project/src/pages/SignUp/SignUp.tsx (hunk 3 above)
Comment (omar-sarfraz) [reply React Project/src/pages/SignUp/SignUp.tsx:61]: I'll add validation here.

File: React Project/src/components/NotFound.tsx (hunk 4 above)
Comment (omar-sarfraz) [reply React Project/src/components/NotFound.tsx:1]: Sure, let me do that.

File: React Project/src/pages/SignUp/SignUp.tsx (hunk 5 above)
Comment (omar-sarfraz) [reply React Project/src/pages/SignUp/SignUp.tsx:81]: Alright. I'll add a custom component.

File: React Project/src/pages/Login/Login.tsx [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+export type LoginError = {
+    type: "" | "EMAIL" | "PASSWORD";
+    message: string;
+};
+
+export default function Login() {
+    const [email, setEmail] = useState<string>("omar@gmail.com");
+    const [password, setPassword] = useState<string>("12345678");
+    const [error, setError] = useState<LoginError>({ type: "", message: "" });
+    const { setUser } = useContext(AuthContext);
```
Comment (larebsyed) [inline React Project/src/pages/Login/Login.tsx:20]: replace with `useAuth`
and `useToast`

**Review by larebsyed — APPROVED** [review]

**Pull request merged**